### ISSUE TITLE
bad partition key when re-keying table

### ISSUE THREAD

**Issue by maccum** [hail-is/hail#3748]
### Hail version:
devel-10a75bb57a6f
### What you did:

```
def test_rekey_correct_partition_key(self):
        ht = hl.utils.range_table(5)
        ht = ht.add_index('a')
        ht = ht.key_by('idx', 'a')
        ht = ht.annotate(b=ht.idx)
        ht = ht.key_by('idx', 'b')
        self.assertEqual(ht.aggregate(agg.sum(ht.idx)), 10)
```

### What went wrong (all error messages here, including the full java stack trace):

```
Java stack trace:
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:156)
	at is.hail.rvd.OrderedRVDType.<init>(OrderedRVDType.scala:21)
	at is.hail.rvd.OrderedRVDType.copy(OrderedRVDType.scala:121)
	at is.hail.expr.TableKeyBy.execute(Relational.scala:1857)
	at is.hail.expr.TableMapRows.execute(Relational.scala:2090)
	at is.hail.expr.TableKeyBy.execute(Relational.scala:1846)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:520)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:39)
	at is.hail.table.Table.aggregate(Table.scala:465)
	at is.hail.table.Table.aggregate(Table.scala:446)
	at is.hail.table.Table.aggregateJSON(Table.scala:436)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by jbloom22** [comment]
I verified that I've fixed this in #3730 .  Will close once it's in.

**Comment by jbloom22** [comment]
And I'll add this test.

**Comment by jbloom22** [comment]
Decided to break out bug fix as separate PR #3749
